be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -56,8 +57,20 @@ Status PipelineDriver::prepare(RuntimeState* runtime_state) {
 
     DCHECK(_state == DriverState::NOT_READY);
 
-    source_operator()->add_morsel_queue(_morsel_queue);
+    auto* source_op = source_operator();
+    // attach ticket_checker to both ScanOperator and SplitMorselQueue
+    auto should_attach_ticket_checker = (dynamic_cast<ScanOperator*>(source_op) != nullptr) &&
```

Comment:
do we support all scan operators currently? did you test it?

Reply:
only OlapScanOperator. only OlapScanOperator can turn on query cache